scripts/us_bjs/nps/preprocess_data.py

```diff
@@ -0,0 +1,222 @@
+"Count_Custody_NotAUSCitizen_Unsentenced", "Count_Custody_Under18_Female", "Count_Custody_Under18_Male", "Count_Custody_Under18", "Count_Jurisdiction_Female", "Count_Jurisdiction_Male", "Count_Jurisdiction"]
+
+DESIRED_COLUMNS = ['GeoId', 'YEAR'] + STAT_VAR_NAMES
+
+def convert_nan_for_calculation(value):
+    if pd.isna(value):
+        return 0
+    else:
+        return value
+
+def get_total_jurisdiction_columns(df):
+    """calculation to include private facility numbers"""
+    df["PVINF_Temp"] = df["PVINF"].apply(convert_nan_for_calculation)
+    df["PVOTHF_Temp"] = df["PVOTHF"].apply(convert_nan_for_calculation)
+    df["PVINM_Temp"] = df["PVINM"].apply(convert_nan_for_calculation)
+    df["PVOTHM_Temp"] = df["PVOTHM"].apply(convert_nan_for_calculation)
+    df["Female_Total_Temp"] = df[["JURTOTF", "PVINF_Temp", "PVOTHF_Temp"]].sum(axis=1).where(df["PVINCLF"] == 2, df["JURTOTF"])
+    df["Male_Total_Temp"] = df[["JURTOTM", "PVINM_Temp", "PVOTHM_Temp"]].sum(axis=1).where(df["PVINCLM"] == 2, df["JURTOTM"])
+    """calculation to include local facility numbers"""
+    df["LFF_Temp"] = df["LFF"].apply(convert_nan_for_calculation)
+    df["LFM_Temp"] = df["LFM"].apply(convert_nan_for_calculation)
+    df["Female_Total_Temp"] = df[["Female_Total_Temp", "LFF_Temp"]].sum(axis=1).where(df["LFINCLF"] == 2, df["Female_Total_Temp"])
+    df["Male_Total_Temp"] = df[["Male_Total_Temp", "LFM_Temp"]].sum(axis=1).where(df["LFINCLM"] == 2, df["Male_Total_Temp"])
+    """calculation to include numbers from local facilities solely to ease crowding"""
+    df["LFCRSTF_Temp"] = df["LFCRSTF"].apply(convert_nan_for_calculation)
+    df["LFCRSTM_Temp"] = df["LFCRSTM"].apply(convert_nan_for_calculation)
+    df["Female_Total_Temp"] = df[["Female_Total_Temp", "LFCRSTF_Temp"]].sum(axis=1).where(df["LFCRINCF"] == 2, df["Female_Total_Temp"])
+    df["Male_Total_Temp"] = df[["Male_Total_Temp", "LFCRSTM_Temp"]].sum(axis=1).where(df["LFCRINCM"] == 2, df["Male_Total_Temp"])
+    """calculation to include federal and other state facility numbers"""
+    df["FEDF_Temp"] = df["FEDF"].apply(convert_nan_for_calculation)
+    df["OTHSTF_Temp"] = df["OTHSTF"].apply(convert_nan_for_calculation)
+    df["FEDM_Temp"] = df["FEDM"].apply(convert_nan_for_calculation)
+    df["OTHSTM_Temp"] = df["OTHSTM"].apply(convert_nan_for_calculation)
+    df["Female_Total_Temp"] = df[["Female_Total_Temp", "FEDF_Temp", "OTHSTF_Temp"]].sum(axis=1).where(df["FACINCLF"] == 2, df["Female_Total_Temp"])
+    df["Male_Total_Temp"] = df[["Male_Total_Temp", "FEDM_Temp", "OTHSTM_Temp"]].sum(axis=1).where(df["FACINCLM"] == 2, df["Male_Total_Temp"])
+    """make jurisdiction total columns"""
+    df["Count_Jurisdiction_Female"] = df["Female_Total_Temp"]
+    df["Count_Jurisdiction_Male"] = df["Male_Total_Temp"]
+    df["Count_Jurisdiction"] = df["Female_Total_Temp"] + df["Male_Total_Temp"]
```

Comment:
nit: wonder if setting these stat-var columns can happen in get_columns. so all stat-var setting happens there.